### PR TITLE
[ci] enforce latest available gem version and use the specific bundler version specified in the lock file

### PR DESCRIPTION
**Pull request by lacostej**
I extracted this old code from #20293. I used it to ensure a more consistent environment across builds.

### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Well now I remember why I did put that in #20293, the appveyor environment fails because we have a too old bundler on appveyor. So we should really merge this before we merge  #20293...

### CODE REVIEW

File: appveyor/bundler_version.rb
Comment (rogerluan) [inline appveyor/bundler_version.rb:1]: Should this file be also in `.ci/` dir? 😊 Despite being used only by appveyor today, I think it's a general CI utility, right?

File: .ci/compatible_gem_version [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+require 'net/http'
+require 'json'
+
+GEM_TO_TEST           = "rubygems-update"
```
Comment (rogerluan) [inline .ci/compatible_gem_version:5]:
```suggestion
GEM_TO_TEST = "rubygems-update"
```

File: .ci/compatible_gem_version [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# Process list to find matching Ruby version
+matching_gem = gem_versions.find { |gem|
+  Gem::Dependency.new('', gem['ruby_version']).
+    match?('', RUBY_VERSION_TO_MATCH)
+}
+
+puts matching_gem['number']
+#puts "Latest version of #{GEM_TO_TEST} " +
+#     "compatible with Ruby #{RUBY_VERSION_TO_MATCH} " +
+#     "is #{matching_gem['number']}."
```
Comment (rogerluan) [inline .ci/compatible_gem_version:27]: Did you intend to leave this here?

**Review by rogerluan — COMMENTED** [review]
Left some questions 🙇

File: .ci/compatible_gem_version (hunk 2 above)
Comment (lacostej) [reply .ci/compatible_gem_version:27]: it was practical to test. I can remove or hide behind a DEBUG.

File: .ci/compatible_gem_version [hunk 3]
```diff
@@ -21,8 +24,4 @@ matching_gem = gem_versions.find { |gem|
     match?('', RUBY_VERSION_TO_MATCH)
 }
 
-puts matching_gem['number']
-#puts "Latest version of #{GEM_TO_TEST} " +
-#     "compatible with Ruby #{RUBY_VERSION_TO_MATCH} " +
-#     "is #{matching_gem['number']}."
-
+puts matching_gem['number']
```
Comment (rogerluan) [inline .ci/compatible_gem_version:27]:
```suggestion
puts matching_gem['number']

```

Missing EOF line break:

image

**Review by rogerluan — APPROVED** [review]
💪

**Pull request merged**